test/e2e/configurable_route_test.go

```diff
@@ -0,0 +1,286 @@
+				RelatedObjects:   []configv1.ObjectReference{{}},
+			},
+		},
+	}
+	if err := kclient.Status().Update(context.TODO(), ingress); err != nil {
+		t.Fatalf("failed to get ingress resource: %v", err)
+	}
+
+	// Check that a role and roleBinding are created for each Spec.ComponentRoutes entry in the openshift-config namespace
+	for _, componentRoute := range ingress.Spec.ComponentRoutes {
+		if err := pollForValidComponentRouteRole(t, componentRoute); err != nil {
+			t.Errorf("expected role unavailable: %v", err)
+		}
+	}
+
+	for _, componentRoute := range ingress.Status.ComponentRoutes {
+		if err := pollForValidComponentRouteRoleBinding(t, componentRoute); err != nil {
+			t.Errorf("expected roleBinding unavailable: %v", err)
+		}
+	}
+
+	// Remove the "bar" consumingUser from the "foo" componentRoute and check that the roleBinding is updated
+	ingress.Status = configv1.IngressStatus{
+		ComponentRoutes: []configv1.ComponentRouteStatus{
+			{
+				Namespace: "default",
+				Name:      "foo",
+				ConsumingUsers: []configv1.ConsumingUser{
+					"system:serviceaccount:default:foo",
+				},
+				DefaultHostname:  "foo.com",
+				CurrentHostnames: []configv1.Hostname{"foo.com"},
+				RelatedObjects:   []configv1.ObjectReference{{}},
+			},
+			{
+				Namespace: "default",
+				Name:      "bar",
+				ConsumingUsers: []configv1.ConsumingUser{
+					"system:serviceaccount:default:bar",
+				},
+				DefaultHostname:  "bar.com",
+				CurrentHostnames: []configv1.Hostname{"bar.com"},
+				RelatedObjects:   []configv1.ObjectReference{{}},
+			},
+		},
+	}
+
+	if err := kclient.Status().Update(context.TODO(), ingress); err != nil {
+		t.Fatalf("failed to get ingress resource: %v", err)
+	}
+
+	for _, componentRoute := range ingress.Status.ComponentRoutes {
+		if err := pollForValidComponentRouteRoleBinding(t, componentRoute); err != nil {
+			t.Errorf("expected roleBinding unavailable: %v", err)
+		}
+	}
+
+	// Remove the `bar` componentRoute from the ingress spec and check that the related role and roleBinding are deleted
+	ingress.Spec.ComponentRoutes = []configv1.ComponentRouteSpec{
+		{
+			Namespace: "default",
+			Name:      "foo",
+			Hostname:  "www.testing.com",
+			ServingCertKeyPairSecret: configv1.SecretNameReference{
+				Name: "foo",
+			},
+		},
+	}
+
+	if err := wait.PollImmediate(1*time.Second, timeout, func() (bool, error) {
+		if err := kclient.Update(context.TODO(), ingress); err != nil {
+			return false, nil
+		}
+		return true, nil
+	}); err != nil {
+		t.Errorf("Failed to update ingress resource: %v", err)
+	}
+
+	// Make sure that the bar role and roleBinding are deleted
+	roleList := &rbacv1.RoleList{}
+	listOptions := []client.ListOption{
+		client.MatchingLabels{
+			componentRouteHashLabelKey: util.Hash("default/bar"),
+		},
+	}
+	if err := wait.PollImmediate(1*time.Second, timeout, func() (bool, error) {
+		if err := kclient.List(context.TODO(), roleList, listOptions...); err != nil {
+			return false, err
```

Comment:
I understand why one would do this with a Get call but a list call that includes no resources should return an empty list rather than an error.

Reply:
We want to prevent transient errors from causing additional test flakiness.  It isn't the job of these tests to tell us if the API or network is flaky; just log the error so we have some clues if the test ultimately fails due to client calls failing, and retry. 
```suggestion
			return false, nil
```